tools/lsp/preview.rs

```diff
@@ -81,10 +129,26 @@ pub fn set_contents(url: &common::VersionedUrl, content: String) {
                 return;
             }
         }
+
+        let Some(path) = common::uri_to_file(&url.url()) else { return Default::default() };
+
+        let fp = path.clone();
+        let fv = url.version().clone();
+        let fc = content.clone();
+
+        let _ = i_slint_core::api::invoke_from_event_loop(move || {
+            let path = fp;
+            let version = fv;
+            let content = fc;
+
+            PREVIEW_STATE
+                .with(move |ps| ps.borrow_mut().refresh_document_cache(&path, version, content))
```

Comment:
would be nice to avoid doing that too often though. Could we do this double parsing only when in design mode for drag and drop. Also not reload all the files but only the file that were changed since.

I'm concerned that this makes the preview reload twice as slow.